src/field/event_cutscene_shaymin.c

```diff
@@ -0,0 +1,110 @@
+#include "overlay_2/event_cutscene.h"
+
+#include "fieldmap.h"
+#include "follow_mon.h"
+#include "math_util.h"
+#include "overlay_01_021F467C.h"
+#include "overlay_01_021F944C.h"
+#include "overlay_01_022053EC.h"
+#include "script_pokemon_util.h"
+#include "task.h"
+#include "unk_0200FA24.h"
+#include "unk_02062108.h"
+#include "unk_020689C8.h"
+static BOOL ov02_022523D0(TaskManager *taskMan);
+
+BOOL ov02_02252334(FieldSystem *fieldSystem) {
+    if (!MapObject_AreBitsSetForMovementScriptInit(PlayerAvatar_GetMapObject(fieldSystem->playerAvatar))) {
+        return FALSE;
+    }
+    if (FollowMon_IsActive(fieldSystem) && fieldSystem->followMon.species == SPECIES_SHAYMIN) {
+        Pokemon *mon = GetFirstAliveMonInParty_CrashIfNone(SaveArray_Party_Get(fieldSystem->saveData));
+        u32 species  = GetMonData(mon, MON_DATA_SPECIES, NULL);
+        u32 form     = GetMonData(mon, MON_DATA_FORM, NULL);
+        GF_ASSERT(species == fieldSystem->followMon.species);
+        // yes, this is explicitly checked twice
+        if (species != fieldSystem->followMon.species) {
+            return FALSE;
+        }
+        if (fieldSystem->followMon.form == SHAYMIN_SKY && form == SHAYMIN_LAND) {
+            return TRUE;
+        }
+    }
+    return FALSE;
+}
+
+void ov02_022523B4(TaskManager *taskMan) {
+    void *data = AllocFromHeapAtEnd(HEAP_ID_FIELD, 12);
+    TaskManager_Call(taskMan, ov02_022523D0, data);
+}
```

Comment:
Move UnkStruct_022523D0 up before this function
```suggestion
void ov02_022523B4(TaskManager *taskMan) {
    UnkStruct_022523D0 *data = AllocFromHeapAtEnd(HEAP_ID_FIELD, sizeof(UnkStruct_022523D0));
    TaskManager_Call(taskMan, ov02_022523D0, data);
}
```